### PR TITLE
feat(todo-14): create API route to edit task

### PR DESCRIPTION
**Pull request by syed-alif-sazim**
**Description of change**
An API route is created to edit a task and update it in the database

**Associated trello link**
https://trello.com/c/HwZPlpYo

### CODE REVIEW

File: src/task/task.controller.ts [hunk 1]
```diff
@@ -21,4 +21,9 @@ export class TaskController {
     await this.taskService.deleteTask(id);
     return { message: 'Task deleted successfully' };
   }
+
+  @Put(':id')
+  async updateTask(@Param('id') id: number, @Body('description') description: string) {
```
Comment (mahir-mohsin-sazim) [inline src/task/task.controller.ts:26]: use DTOs for the payload validation

File: src/task/task.service.ts [hunk 2]
```diff
@@ -33,4 +33,16 @@ export class TaskService {
 
     await this.taskRepository.getEntityManager().removeAndFlush(task);
   }
+
+  async updateTask(id: number, description: string): Promise<Task> {
+    const task = await this.taskRepository.findOne({ id });
+    if (!task) {
+      throw new Error('Task not found');
```
Comment (mahir-mohsin-sazim) [inline src/task/task.service.ts:40]: use NestJS built in `Exception handlers` please

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]

File: src/task/task.controller.ts (hunk 1 above)
Comment (syed-alif-sazim) [reply src/task/task.controller.ts:26]: I resolved this in the last PR

File: src/task/task.service.ts (hunk 2 above)
Comment (syed-alif-sazim) [reply src/task/task.service.ts:40]: I resolved this in the last PR

**Review by mahir-mohsin-sazim — APPROVED** [review]

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**